src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -132,63 +132,20 @@ private function preRender(MountedComponent $mounted, array $context = []): PreR
             // add attributes
             [$metadata->getAttributesVar() => $mounted->getAttributes()],
         );
-        $event = new PreRenderEvent($mounted, $metadata, $variables);
 
+        // TODO pass only props & vars here
+        // --> no context, outerscope, magic, ..
+
+        $event = new PreRenderEvent($mounted, $metadata, $variables);
         $this->dispatcher->dispatch($event);
 
         return $event;
     }
 
-    private function exposedVariables(object $component, bool $exposePublicProps): \Iterator
+    public function warmup(): array
```

Comment:
I don't think this method is used, right?

Reply:
Oops 😅  removed, thx !